src/java/nrepl/QuotaWriter.java

```diff
@@ -0,0 +1,94 @@
+package nrepl;
+
+import java.io.*;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * Wrapper around <code>java.io.Writer</code> that throws
+ * <code>nrepl.QuotaExceeded</code> once it has written more than quota bytes.
+ */
+public class QuotaWriter extends Writer {
```

Comment:
Or perhaps WriterWithQuota. Hard to find the optimal name for this one, but the current name will also do.

Reply:
QuotaBoundWriter?

Reply:
Love this name!